File: legacy/sdk/typescript/src/resources/agents.ts
/** Agents resource. */
import type { Axion } from "../client.js";
import type { AgentInstance, AgentTemplate, Task } from "../types.js";

export interface HireOptions {
  template: string;
  name: string;
  config?: Record<string, unknown>;
  enabledTools?: string[];
  customPrompt?: string;
  voiceCloneId?: string;
  budgetPerDay?: number;
}

export class AgentsResource {
  constructor(private readonly _c: Axion) {}

  catalog(filter?: { category?: string; q?: string }): Promise<AgentTemplate[]> {
    return this._c._request<AgentTemplate[]>("GET", "/agents/catalog", {
      query: filter ?? {},
    });
  }

  template(slug: string): Promise<AgentTemplate> {
    return this._c._request<AgentTemplate>("GET", `/agents/catalog/${slug}`);
  }

  list(opts?: { status?: string }): Promise<AgentInstance[]> {
    return this._c._request<AgentInstance[]>("GET", "/agents", { query: opts });
  }

  get(agentId: string): Promise<AgentInstance> {
    return this._c._request<AgentInstance>("GET", `/agents/${agentId}`);
  }

  hire(opts: HireOptions): Promise<AgentInstance> {
    return this._c._request<AgentInstance>("POST", "/agents", {
      body: {
        template_slug: opts.template,
        name: opts.name,
        config: opts.config ?? {},
        enabled_tools: opts.enabledTools ?? [],
        custom_prompt: opts.customPrompt,
        voice_clone_id: opts.voiceCloneId,
        budget_per_day_eur: opts.budgetPerDay ?? 100,
      },
    });
  }

  update(agentId: string, fields: Partial<AgentInstance>): Promise<AgentInstance> {
    return this._c._request<AgentInstance>("PATCH", `/agents/${agentId}`, {
      body: fields,
    });
  }

  async delete(agentId: string): Promise<void> {
    await this._c._request("DELETE", `/agents/${agentId}`);
  }

  async pause(agentId: string): Promise<void> {
    await this._c._request("POST", `/agents/${agentId}/pause`);
  }

  async resume(agentId: string): Promise<void> {
    await this._c._request("POST", `/agents/${agentId}/resume`);
  }

  run(
    agentId: string,
    opts: { objective: string; inputs?: Record<string, unknown>; timeoutS?: number }
  ): Promise<Task> {
    return this._c._request<Task>("POST", `/agents/${agentId}/run`, {
      body: {
        objective: opts.objective,
        inputs: opts.inputs ?? {},
        timeout_s: opts.timeoutS ?? 300,
      },
    });
  }
}

File: legacy/sdk/python/axion/resources/agents.py
"""Agents resource."""
from __future__ import annotations

from typing import TYPE_CHECKING, Any

if TYPE_CHECKING:
    from axion.client import Axion


class AgentsResource:
    def __init__(self, client: "Axion") -> None:
        self._c = client

    # ── Catalog ───────────────────────────────────────────────

    def catalog(self, *, category: str | None = None, q: str | None = None) -> list[dict]:
        params = {k: v for k, v in {"category": category, "q": q}.items() if v}
        return self._c._request("GET", "/agents/catalog", params=params)

    def template(self, slug: str) -> dict:
        return self._c._request("GET", f"/agents/catalog/{slug}")

    # ── Instances ────────────────────────────────────────────

    def list(self, *, status: str | None = None) -> list[dict]:
        params = {"status": status} if status else {}
        return self._c._request("GET", "/agents", params=params)

    def get(self, agent_id: str) -> dict:
        return self._c._request("GET", f"/agents/{agent_id}")

    def hire(
        self,
        *,
        template: str,
        name: str,
        config: dict[str, Any] | None = None,
        enabled_tools: list[str] | None = None,
        custom_prompt: str | None = None,
        voice_clone_id: str | None = None,
        budget_per_day: int = 100,
    ) -> dict:
        body = {
            "template_slug": template,
            "name": name,
            "config": config or {},
            "enabled_tools": enabled_tools or [],
            "custom_prompt": custom_prompt,
            "voice_clone_id": voice_clone_id,
            "budget_per_day_eur": budget_per_day,
        }
        return self._c._request("POST", "/agents", json=body)

    def update(self, agent_id: str, **fields: Any) -> dict:
        return self._c._request("PATCH", f"/agents/{agent_id}", json=fields)

    def delete(self, agent_id: str) -> None:
        self._c._request("DELETE", f"/agents/{agent_id}")

    def pause(self, agent_id: str) -> None:
        self._c._request("POST", f"/agents/{agent_id}/pause")

    def resume(self, agent_id: str) -> None:
        self._c._request("POST", f"/agents/{agent_id}/resume")

    def run(
        self,
        agent_id: str,
        *,
        objective: str,
        inputs: dict[str, Any] | None = None,
        timeout_s: int = 300,
    ) -> dict:
        """Trigger an ad-hoc run. Returns the Task object."""
        return self._c._request(
            "POST",
            f"/agents/{agent_id}/run",
            json={
                "objective": objective,
                "inputs": inputs or {},
                "timeout_s": timeout_s,
            },
        )
